api/service.go

```diff
@@ -524,6 +528,158 @@ func (s *service) RemoveAppToken(ctx context.Context, req *pb.RemoveAppTokenRequ
 	return &pb.RemoveAppTokenReply{}, nil
 }
 
+// AddFolder handles an add folder request.
+func (s *service) AddFolder(ctx context.Context, req *pb.AddFolderRequest) (*pb.AddFolderReply, error) {
+	log.Debugf("received add folder request")
+
+	scope, ok := ctx.Value(reqKey("scope")).(string)
+	if !ok {
+		log.Fatal("scope required")
+	}
+	proj, err := s.getProjectForScope(ctx, req.ProjectID, scope)
+	if err != nil {
+		return nil, err
+	}
+
+	pth, err := s.ipfsClient.Unixfs().Add(
+		ctx,
+		ipfsfiles.NewMapDirectory(map[string]ipfsfiles.Node{}),
```

Comment:
Here we just create an empty directory.

Reply:
Yes, this will always return the same node since it's just an empty directory. Below, we pin it, but it won't be pinned again if it's already pinned, say, by another project. However, we should ensure that it's not _unpinned_ since we don't want to unpin if another project is relying on it. In theory, this applies to any directory... perhaps we should explore some way of making all directories not conflict across projects. There's a non-zero chance that you and I create the exact same directory (name and contents).

Reply:
Oh, I see. So this operation always returns the same empty folder for all _starting_ new folders.
But as soon as they add files, that _node_ would be _mutated_ and will start becoming an _onwed noded_ of the project.  I guess the `Pin().Update(..)` call to move the pinning wouldn't hurt. Since the next created folder will create this empty directory again. (I hope this makes sense).

Regarding creating identical graphs in different folders, maybe instead of creating an empty folder to start, it can have a file with some random content, and that file is always ignored in listings or can't be removed. That would make the root cid always unique per project-folder. Something like creating a `.git` when you do `git init` (first idea in my mind).

Reply:
Nice, I like that! I'll create an issue.